### ISSUE TITLE
Error: Can't define Custom locations - status "Offline" - v2.11.1 (latest)

### ISSUE THREAD

**Issue by Gh0stExp10it** [NginxProxyManager/nginx-proxy-manager#3484]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**
After I've upgraded to the latest version v2.11.1, I can't define ```Cusom locations``` for a ```Proxy Host```.  The status on the dashboard changes from *Online*&#128994; to *Offline*&#x1F534;.
The previously existing proxy host configuration file is deleted, but not recreated (e.g.: /data/nginx/proxy_host/1.conf).

The output of the container log after pressing the "Save" button on the "Edit Proxy Host" shows the following:
```sh
[1/21/2024] [4:59:40 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
[1/21/2024] [4:59:40 PM] [Nginx    ] › ⬤  debug     Deleting file: /data/nginx/proxy_host/1.conf
[1/21/2024] [4:59:40 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
[1/21/2024] [4:59:40 PM] [Nginx    ] › ⬤  debug     Deleting file: /data/nginx/proxy_host/1.conf
[1/21/2024] [4:59:40 PM] [Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "code": "ENOENT",
  "syscall": "unlink",
  "path": "/data/nginx/proxy_host/1.conf"
}
[1/21/2024] [4:59:40 PM] [Nginx    ] › ⬤  debug     Deleting file: /data/nginx/proxy_host/1.conf.err
[1/21/2024] [4:59:40 PM] [Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "code": "ENOENT",
  "syscall": "unlink",
  "path": "/data/nginx/proxy_host/1.conf.err"
}
[1/21/2024] [4:59:40 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -t -g "error_log off;"
[1/21/2024] [4:59:40 PM] [Nginx    ] › ℹ  info      Reloading Nginx
[1/21/2024] [4:59:40 PM] [Global   ] › ⬤  debug     CMD: /usr/sbin/nginx -s reload
```

**Nginx Proxy Manager Version**
v2.11.1 (latest, as of date 21.01.2024)

**To Reproduce**
1. Go to "Hosts" and "Proxy Hosts"
2. Click on the three-dotted menu
3. Click on edit
4. Switch to the tab "Custom locations"
5. Create a new Custom location (example see screenshot below)
6. Click on the button "Save"
7. The status should be now switched to *Offline*

**Expected behavior**
It should be possible to define a custom location without deleting the proxy conf and thus taking the host offline.
In version v2.10.4 this is possible without any problems. I was able to test this successfully by rolling back to this version.

**Screenshots**
<img width="248" alt="screenshot-npm-custom-locations-01" src="https://github.com/NginxProxyManager/nginx-proxy-manager/assets/7295005/96a2f2c9-0f15-43d6-b543-9338d59eadcc">
<img width="113" alt="screenshot-npm-custom-locations-02" src="https://github.com/NginxProxyManager/nginx-proxy-manager/assets/7295005/7671e46c-a5a6-4486-84f9-a1d63809d1e0">

**Operating System**
Ubuntu Server 22.04.3 LTS

**Additional context**
/None/

**Comment by tanpro260196** [comment]
Most like a duplicate of this: https://github.com/NginxProxyManager/nginx-proxy-manager/issues/3474
TLDR: Using HSTS and custom location at the same time break the config.

**Comment by tobilektri** [comment]
Same here:

```
2024/01/22 10:35:02 [emerg] 747#747: "map" directive is not allowed here in /data/nginx/proxy_host/10.conf:47
2024/01/22 10:35:02 [notice] 751#751: signal process started
```

**Comment by Kavan72** [comment]
Getting same error.

**Comment by RichLii** [comment]
roll back to 2.10.4 can solve this problem

**Comment by Gh0stExp10it** [comment]
> Most likely a duplicate of this: #3474 TLDR: Using HSTS and custom location at the same time break the config.

Thanks for the additional info.
Unfortunately, this is not the case, HSTS is switched off/deactivated.

**Comment by innosflew** [comment]
I'm experiencing the same issue, and I don't use HSTS either.

**Comment by bleak183** [comment]
well it seems the project is pretty much abandoned but for whatever it is worth, I am getting the same issue

**Comment by bleak183** [comment]
> roll back to 2.10.4 can solve this problem

I can confirm this works

**Comment by ramkrishna757575** [comment]
Same here...suddenly all my reverse proxies broke due to this

**Comment by jpdsc** [comment]
Same here, there is a workaround in https://github.com/NginxProxyManager/nginx-proxy-manager/issues/3474 but will wait until resolved and reverted to :2.10.4 for now.

**Comment by mbah-dukun** [comment]
same here . . . .
is like npm not have permission for execute folder map

**Comment by daraHK88** [comment]
i have the same problem

**Comment by TomRobinsonIO** [comment]
Same issue here, HSTS on/off does not make a difference.

**Comment by MattXcz** [comment]
Same issue:

`nginx: [warn] the &quot;listen ... http2&quot; directive is deprecated, use the &quot;http2&quot; directive instead in /data/nginx/proxy_host/2.conf:19
nginx: [warn] the &quot;listen ... http2&quot; directive is deprecated, use the &quot;http2&quot; directive instead in /data/nginx/proxy_host/2.conf:20
nginx: [warn] the &quot;listen ... http2&quot; directive is deprecated, use the &quot;http2&quot; directive instead in /data/nginx/proxy_host/4.conf:19
nginx: [warn] protocol options redefined for 0.0.0.0:443 in /data/nginx/proxy_host/4.conf:19
nginx: [warn] the &quot;listen ... http2&quot; directive is deprecated, use the &quot;http2&quot; directive instead in /data/nginx/proxy_host/4.conf:20
nginx: [warn] protocol options redefined for [::]:443 in /data/nginx/proxy_host/4.conf:20
nginx: [emerg] &quot;map&quot; directive is not allowed here in /data/nginx/proxy_host/4.conf:65
nginx: configuration file /usr/local/openresty/nginx/conf/nginx.conf test failed
`

**Comment by Mkeefeus** [comment]
Same issue here as well, HSTS turned off and still unable to start with custom locations

**Comment by liukonen** [comment]
Using "Latest" branch... Switching boxes because my services just stopped working... adding subdomains, and It goes to offline once the subdomain is added. HSTS does not matter

[2/2/2024] [2:48:15 AM] [Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "code": "ENOENT",
  "syscall": "unlink",
  "path": "/data/nginx/proxy_host/4.conf.err"
}

**Comment by ramkrishna757575** [comment]
I have ditched npm. Vanilla nginx is da wey

**Comment by AkshayRao27** [comment]
See if this solution works for you: https://github.com/NginxProxyManager/nginx-proxy-manager/issues/3512#issuecomment-1940868886
